updated execScriptAsync

juliantheberge opened the pull request:
## Pull Request for EasyDBI

### Purpose
- execScriptAsync and loadScriptAsync in lib/driver do not parse queries from an .sql file correctly
- Tl;dr: change queries definition in above methods to
```
var queries = data.replace(/(--).*/, '').split(/\s*;\s*/);
```

- And add falsy check in return map
```
if (query) { return this.execAsync(query) }
```

### Issue One: Regex
Currently both methods read the file and split the output into queries via regex, then each query is run as a promise
```
let queries = data.split(/\w*;\w/)
```
This doesn’t work for the following reasons.
`<\w>` matches any word character (equal to `<[a-zA-Z0-9_]>`) but doesn’t capture whitespace, which is a common pattern for separating queries in sql files.
E.g. something like this
```
create table users 
     _id integer primary key not null,
     user_uuid varchar(36) unique not null
);

create table auth_tokens (
    _id integer primary key not null,
    user_uuid varchar(36) references users ( user_uuid ) not null,
    session_id varchar(36) unique not null
);
```
Even in the event that the two queries have no whitespace separating them, the `<;\w>` pattern will split at the semicolon and then next char, removing them both.
```
'…);create…' => [‘...)’, ‘reate…’]
```

### Issue Two: Empty Query
Because the split statment breaks at the above regex it leaves behind one empty string space at the end of the array `<[ … , ‘’]`>
To filter for this empty query add a falsy check in the map
```
if (query) {
	return _this.execAsync(query);
}
```
### Issue Three: Comments

Even with the updated parse, comments are treated as queries.
Remove comments with a replace statement
```
data.replace(/(--).*/, '');
```

juliantheberge committed:
updated execScriptAsync
- files now divided into queries correctly with new regex
- comments no longer treated as queries
- added check for empty string queries

juliantheberge committed:
added note to readme about using require for drivers

yinso reviewed (changes requested):
Per our conversation - please remove the JavaScript that got changed without corresponding changed TypeScript files.

juliantheberge committed:
js files removed, g flag added to replace regex in lib/driver

juliantheberge committed:
inclided compiling for exec-file tests and driver tests, as well as readme.md

juliantheberge committed:
included compiling for exec-file tests and driver tests, as well as readme.md

juliantheberge committed:
Merge branch 'parse-sql-file' of https://github.com/thelionthatiam/easydbi into parse-sql-file

juliantheberge committed:
merged /g flag changes to regex

juliantheberge committed:
- added specificity to query split regex test
- added specificity to remove comments regex replace
- added util to hold regex functions
- added .js, .d.ts, and .js.map where applicable

juliantheberge committed:
added original compiled javascript versions of lib files

juliantheberge committed:
homogenized newline for testing on uinx and windows in test/exec-file

yinso reviewed (approved):
Looks great thanks!

juliantheberge committed:
Merge remote-tracking branch 'upstream/master' into parse-sql-file
